File: Data Structures & Algorithms/maximum-subarray/submission-1.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        count = 0
        maxcount = nums[0]

        for num in nums:
            if count < 0:
                count = 0

            count += num
            maxcount = max(maxcount , count)

        
        return maxcount

File: Data Structures & Algorithms/maximum-product-subarray/submission-1.cpp
class Solution {
public:
    int maxProduct(vector<int>& nums) {
        int res = nums[0];
        int mn =1 , mx =1;

        for(auto num: nums){
            int tmp = mx * num;

            mx = max(num*mx , max(num*mn , num));
            mn = min(tmp , min(num*mn , num));
            res = max(res, mx);
        }
        return res;
    }
};
